### PR TITLE
feat(robot-server): split database into v5 and v6

### PR DESCRIPTION
**Pull request by sanni-t**
# Overview

Splits up the changes we have so far into a v5 schema that's already in QA and anything on top of that to be in v6 so that QA & ABR don't have to do database resets while testing.

This branch is currently pointing to edge that has the v5 schema that's in QA. This is done on purpose so that we can check that the database changes diff between current edge (without v5.1) and introduction of v6 is correct and is not affected by the intermediate v5.1 DB changes.

Only look at the changes in the persistence directory (and its tests). Everything else is part of the outstanding PRs that introduced v5.1 changes and is irrelevant to the matter at hand.

Will rebase the changes onto edge once edge has the 5.1 changes.

### CODE REVIEW

File: robot-server/robot_server/persistence/_migrations/v5_to_v6.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            dest_engine = exit_stack.enter_context(sql_engine_ctx(dest_db_file))
+            schema_6.metadata.create_all(dest_engine)
+
+            source_transaction = exit_stack.enter_context(source_engine.begin())
+            dest_transaction = exit_stack.enter_context(dest_engine.begin())
+
+            _migrate_db_with_changes(source_transaction, dest_transaction)
+
+
+def _migrate_db_with_changes(
```
Comment (SyntaxColoring) [inline robot-server/robot_server/persistence/_migrations/v5_to_v6.py:68]: Is this intentionally missing the `data_files` table?

Comment (sanni-t) [reply robot-server/robot_server/persistence/_migrations/v5_to_v6.py:68]: Nope. Good catch.

**Review by SyntaxColoring — APPROVED** [review]
Looks good to me except for the potentially missing `data_files` table.

**Comment by SyntaxColoring** [comment]
Oh and also the directory of data files themselves? I’m on my phone and having trouble bouncing around the code, but it looks like the file contents live outside the table?

**Comment by sanni-t** [comment]
> Oh and also the directory of data files themselves? I’m on my phone and having trouble bouncing around the code, but it looks like the file contents live outside the table?

Yep! Thanks for catching that!
I really shouldn't code while sick

**Review by SyntaxColoring — APPROVED** [review]
Looks great whenever you’re ready to merge. And get some rest!

**Pull request merged**